### PR TITLE
Adding some legal validations during the flow

### PR DESCRIPTION
**Pull request by Poppe-A**
#7743

### CODE REVIEW

File: apps/catalog/catalog/src/app/dashboard/sales/shell/view/view.component.html [hunk 1]
```diff
@@ -2,23 +2,28 @@
...
+              <span>Accept</span>
             </button>
-        </ng-template>
-        <a [disabled]="negotiation.createdByOrg === orgId" mat-stroked-button color="primary" routerLink="../negotiation">
-          <mat-icon svgIcon="negotiate"></mat-icon>
-          <span>Negotiate</span>
-        </a>
+          </ng-container>
+          <ng-template #noPrice>
+              <button disabled mat-stroked-button>
```
Comment (RemcoSimonides) [inline apps/catalog/catalog/src/app/dashboard/sales/shell/view/view.component.html:17]: Fix indentation

File: libs/contract/src/lib/contract/components/confirm-decline/confirm-decline.component.scss [hunk 2]
```diff
@@ -11,3 +11,7 @@ mat-spinner {
   margin-left: 8px;
   display: inline-block;
 }
+
+.validation-checkbox {
+  margin-block: 24px;
```
Comment (RemcoSimonides) [inline libs/contract/src/lib/contract/components/confirm-decline/confirm-decline.component.scss:16]: Normal `margin` wasn't what you want?

File: libs/contract/src/lib/contract/components/confirm-decline/confirm-decline.component.html [hunk 3]
```diff
@@ -18,8 +18,14 @@ <h1>Please give a reason for your refusal.</h1>
     </textarea>
     <mat-hint>Please specify why you want to decline this offer.</mat-hint>
   </mat-form-field>
+  <div *ngIf="data.validationCheckbox" class="validation-checkbox" fxLayout="column" fxLayoutAlign="center center">
+    <p><strong>Before you accept this contract, please agree on the following:</strong></p>
+    <mat-checkbox formControlName="acceptTerms">
+      I hereby accept the policy regarding accepted offers, as mentioned in the <a routerLink="toDefine" target="_blank">Terms and Conditions.</a>
```
Comment (RemcoSimonides) [inline libs/contract/src/lib/contract/components/confirm-decline/confirm-decline.component.html:24]: Add `<!-- TODO issue number -->`
for easier reference later.

File: libs/contract/src/lib/contract/components/confirm-decline/confirm-decline.component.ts [hunk 4]
```diff
@@ -4,7 +4,10 @@ import { ChangeDetectionStrategy, Component, Inject } from '@angular/core';
 import { FormControl, FormGroup } from '@angular/forms';
 import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
 
-export interface ConfirmDeclineData { type: 'seller' | 'buyer' }
+export interface ConfirmDeclineData { 
+  type: 'seller' | 'buyer', 
+  validationCheckbox?: boolean
```
Comment (RemcoSimonides) [inline libs/contract/src/lib/contract/components/confirm-decline/confirm-decline.component.ts:9]: I think this should be renamed to something like `showAcceptTermsCheckbox`. Francois doesn't like long names, but I think it's more explicit. With just `validationCheckbox` it isn't fully clear what kind of checkbox we're talking about if I don't know the UI/code

File: libs/ui/src/lib/confirm/confirm.component.html [hunk 5]
```diff
@@ -10,8 +10,14 @@
     <h2>{{ data.title || "Confirm" }}</h2>
     <p>{{ data.question || "Do you confirm ?" }}</p>
   </div>
+  <form *ngIf="data.validationCheckbox" [formGroup]="form" fxLayout="column" fxLayoutAlign="center center">
+    <p><strong>Before you accept this contract, please agree on the following:</strong></p>
+    <mat-checkbox formControlName="acceptTerms">
+      I hereby accept the policy regarding accepted offers, as mentioned in the <a routerLink="toDefine" target="_blank">Terms and Conditions.</a>
```
Comment (RemcoSimonides) [inline libs/ui/src/lib/confirm/confirm.component.html:16]: Add `<!-- TODO issue number-->`
for easier reference later.

File: libs/ui/src/lib/confirm/confirm.component.ts [hunk 6]
```diff
@@ -9,26 +10,31 @@ import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
...
-        question: string,
-        confirm: string,
-        cancel: string,
-        onConfirm?: () => void
-      },
-      public dialogRef: MatDialogRef<ConfirmComponent>,
-    ) { }
+  form = new FormGroup({
+    acceptTerms: new FormControl(false)
+  });
```
Comment (RemcoSimonides) [inline libs/ui/src/lib/confirm/confirm.component.ts:15]: No need to make a `FormGroup` if you only need 1 control.
`acceptTermsControl = new FormControl(false)` is enough.
Then in the HTML you also don't need to wrap the control with `<form>`

Comment (RemcoSimonides) [reply libs/ui/src/lib/confirm/confirm.component.ts:15]: If you expect more controls later, It's fine to keep it like this imo

File: libs/ui/src/lib/confirm/confirm.component.ts [hunk 7]
```diff
@@ -1,4 +1,5 @@
 import { Component, ChangeDetectionStrategy, Inject } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
```
Comment (RemcoSimonides) [inline libs/ui/src/lib/confirm/confirm.component.ts:2]: Unused import

**Comment by Poppe-A** [comment]
!! Code is finished, just waiting for #7938 to be ready before merging

File: libs/ui/src/lib/confirm-with-validation/confirm-with-validation.component.html [hunk 8]
```diff
@@ -0,0 +1,28 @@
+
```
Comment (RemcoSimonides) [inline libs/ui/src/lib/confirm-with-validation/confirm-with-validation.component.html:1]: Remove empty line

File: libs/ui/src/lib/confirm-with-validation/confirm-with-validation.component.ts [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+      question: string,
+      confirm: string,
+      cancel: string,
+      onConfirm?: () => void
+    },
+    public router: Router,
+    public dialogRef: MatDialogRef<ConfirmWithValidationComponent>,
+  ) { }
+
+  public termsPath = `/c/o/${this.router.url.includes('marketplace') ? 'marketplace' : 'dashboard'}/terms`;
```
Comment (RemcoSimonides) [inline libs/ui/src/lib/confirm-with-validation/confirm-with-validation.component.ts:28]: Same as above

File: libs/ui/src/lib/confirm-with-validation/confirm-with-validation.module.ts [hunk 10]
```diff
@@ -0,0 +1,24 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { ConfirmWithValidationComponent } from './confirm-with-validation.component';
+import { MatButtonModule } from '@angular/material/button';
+import { FlexLayoutModule } from '@angular/flex-layout';
+import { MatIconModule } from '@angular/material/icon';
+import { MatCheckboxModule } from '@angular/material/checkbox';
+import { RouterModule } from '@angular/router';
+import { ReactiveFormsModule } from '@angular/forms';
```
Comment (RemcoSimonides) [inline libs/ui/src/lib/confirm-with-validation/confirm-with-validation.module.ts:9]: Please sort it nicely as we usually do with many imports

File: libs/contract/src/lib/contract/components/confirm-decline/confirm-decline.component.ts [hunk 11]
```diff
@@ -29,10 +33,14 @@ export class ConfirmDeclineComponent {
   form = new FormGroup({
     reason: new FormControl(''),
     message: new FormControl(''),
+    acceptTerms: new FormControl(false)
   });
 
+  termsPath = `/c/o/${this.router.url.includes('marketplace') ? 'marketplace' : 'dashboard'}/terms`;
```
Comment (RemcoSimonides) [inline libs/contract/src/lib/contract/components/confirm-decline/confirm-decline.component.ts:39]: You didn't like my suggestion @Poppe-A ? With
`termsPath = this.router.url.includes('marketplace') ? '/c/o/marketplace/terms' : '/c/o/dashboard/terms';`
For readability

File: libs/contract/src/lib/contract/components/confirm-with-validation/confirm-with-validation.component.ts [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+@Component({
+  selector: 'blockframes-confirm-validate',
+  templateUrl: './confirm-with-validation.component.html',
+  styleUrls: ['./confirm-with-validation.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class ConfirmWithValidationComponent {
+
+  acceptTerms = new FormControl(false);
+  termsPath = this.router.url.includes('marketplace') ? '/c/o/marketplace/terms' : '/c/o/dashboard/terms';
```
Comment (phpgeekfr) [inline libs/contract/src/lib/contract/components/confirm-with-validation/confirm-with-validation.component.ts:15]: same

**Review by phpgeekfr — APPROVED** [review]

**Pull request merged**